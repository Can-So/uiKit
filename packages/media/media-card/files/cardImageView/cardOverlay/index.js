import * as tslib_1 from "tslib";
import * as React from 'react';
import { Component } from 'react';
import { FormattedMessage } from 'react-intl';
import * as cx from 'classnames';
import TickIcon from '@findable/icon/glyph/check';
import { Ellipsify } from '@findable/media-ui';
import { messages } from '@findable/media-ui';
// We dont require things directly from "utils" to avoid circular dependencies
import { FileIcon } from '../../../utils/fileIcon';
import { ErrorIcon } from '../../../utils/errorIcon';
import CardActions from '../../../utils/cardActions';
import { TickBox, Overlay, ErrorLine, LeftColumn, TopRow, BottomRow, RightColumn, ErrorMessage, Retry, TitleWrapper, Subtitle, Metadata, ErrorWrapper, } from './styled';
var CardOverlay = /** @class */ (function (_super) {
    tslib_1.__extends(CardOverlay, _super);
    function CardOverlay(props) {
        var _this = _super.call(this, props) || this;
        _this.onMenuToggle = function (attrs) {
            _this.setState({ isMenuExpanded: attrs.isOpen });
        };
        _this.state = {
            isMenuExpanded: false,
        };
        return _this;
    }
    Object.defineProperty(CardOverlay.prototype, "wrapperClassNames", {
        get: function () {
            var _a = this.props, error = _a.error, noHover = _a.noHover, selectable = _a.selectable, selected = _a.selected, mediaType = _a.mediaType, persistent = _a.persistent;
            var isMenuExpanded = this.state.isMenuExpanded;
            return error
                ? cx('overlay', { error: error, active: isMenuExpanded })
                : cx('overlay', mediaType, {
                    active: isMenuExpanded,
                    selectable: selectable,
                    selected: selected,
                    // Yes, you right. We put "persistent" class when it is NOT persistent. ðŸ¤¦
                    persistent: !persistent,
                    noHover: noHover,
                });
        },
        enumerable: true,
        configurable: true
    });
    CardOverlay.prototype.render = function () {
        var _a = this.props, error = _a.error, noHover = _a.noHover, mediaName = _a.mediaName, persistent = _a.persistent, actions = _a.actions;
        var titleText = error || !mediaName ? '' : mediaName;
        var menuTriggerColor = !persistent ? 'white' : undefined;
        return (React.createElement(Overlay, { hasError: !!error, noHover: noHover, className: this.wrapperClassNames },
            React.createElement(TopRow, { className: 'top-row' },
                this.errorLine(),
                React.createElement(TitleWrapper, { className: 'title' },
                    React.createElement(Ellipsify, { text: titleText, lines: 2 })),
                this.tickBox()),
            React.createElement(BottomRow, { className: 'bottom-row' },
                React.createElement(LeftColumn, null, this.bottomLeftColumn()),
                React.createElement(RightColumn, null, actions ? (React.createElement(CardActions, { actions: actions, onToggle: this.onMenuToggle, triggerColor: menuTriggerColor })) : null))));
    };
    CardOverlay.prototype.errorLine = function () {
        var error = this.props.error;
        return (error && (React.createElement(ErrorLine, null,
            React.createElement(ErrorMessage, null, this.props.error))));
    };
    CardOverlay.prototype.tickBox = function () {
        var _a = this.props, selected = _a.selected, selectable = _a.selectable;
        var tick = React.createElement(TickIcon, { label: "tick" });
        var className = cx('tickbox', { selected: selected });
        return selectable && React.createElement(TickBox, { className: className },
            " ",
            tick,
            " ");
    };
    CardOverlay.prototype.bottomLeftColumn = function () {
        var _a = this.props, error = _a.error, onRetry = _a.onRetry;
        if (error) {
            if (!onRetry) {
                return React.createElement(ErrorIcon, null);
            }
            return (React.createElement(ErrorWrapper, null,
                React.createElement(ErrorIcon, null),
                React.createElement(Retry, { onClick: onRetry },
                    React.createElement(FormattedMessage, tslib_1.__assign({}, messages.retry)))));
        }
        else {
            var _b = this.props, mediaType = _b.mediaType, subtitle = _b.subtitle, icon = _b.icon;
            var classNames_1 = cx('metadata');
            var fileIcon = mediaType || icon ? (React.createElement(FileIcon, { mediaType: mediaType, iconUrl: icon })) : null;
            var subtitleEl = subtitle ? (React.createElement(Subtitle, { className: "file-size" }, subtitle)) : null;
            return (React.createElement("div", null,
                React.createElement(Metadata, { className: classNames_1 },
                    fileIcon,
                    subtitleEl)));
        }
    };
    CardOverlay.prototype.removeBtnClick = function (handler) {
        return function (e) {
            e.preventDefault();
            e.stopPropagation();
            handler();
        };
    };
    CardOverlay.defaultProps = {
        actions: [],
        mediaName: '',
    };
    return CardOverlay;
}(Component));
export { CardOverlay };
//# sourceMappingURL=index.js.map