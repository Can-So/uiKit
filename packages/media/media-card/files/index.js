export * from './card';
export * from './cardImageView';
//# sourceMappingURL=index.js.map