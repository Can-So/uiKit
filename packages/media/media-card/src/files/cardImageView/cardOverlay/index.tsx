import * as React from 'react';
import { MouseEvent, Component, ReactNode } from 'react';
import { FormattedMessage } from 'react-intl';
import * as cx from 'classnames';
import { MediaType } from '@atlaskit/media-core';
import TickIcon from '@atlaskit/icon/glyph/check';
import { Ellipsify } from '@atlaskit/media-ui';
import { messages } from '@atlaskit/media-ui';
// We dont require things directly from "utils" to avoid circular dependencies
import { FileIcon } from '../../../utils/fileIcon';
import { ErrorIcon } from '../../../utils/errorIcon';
import CardActions from '../../../utils/cardActions';
import { CardAction, CardEventHandler } from '../../../actions';

import {
  TickBox,
  Overlay,
  ErrorLine,
  LeftColumn,
  TopRow,
  BottomRow,
  RightColumn,
  ErrorMessage,
  Retry,
  TitleWrapper,
  Subtitle,
  Metadata,
  ErrorWrapper,
} from './styled';

export interface CardOverlayProps {
  mediaType?: MediaType;
  mediaName?: string;
  subtitle?: string;

  selectable?: boolean;
  selected?: boolean;
  persistent: boolean;

  error?: ReactNode;
  onRetry?: () => void;

  actions?: Array<CardAction>;
  icon?: string;
}

export interface CardOverlayState {
  isMenuExpanded: boolean;
}

export class CardOverlay extends Component<CardOverlayProps, CardOverlayState> {
  static defaultProps = {
    actions: [],
    mediaName: '',
  };

  constructor(props: CardOverlayProps) {
    super(props);

    this.state = {
      isMenuExpanded: false,
    };
  }

  private get wrapperClassNames() {
    const { error, selectable, selected, mediaType, persistent } = this.props;
    const { isMenuExpanded } = this.state;

    return error
      ? cx('overlay', { error, active: isMenuExpanded })
      : cx('overlay', mediaType, {
          active: isMenuExpanded,
          selectable,
          selected,
          persistent: !persistent,
        });
  }

  render() {
    const { error, mediaName, persistent, actions } = this.props;
    const titleText = error || !mediaName ? '' : mediaName;
    const menuTriggerColor = !persistent ? 'white' : undefined;

    return (
      <Overlay hasError={!!error} className={this.wrapperClassNames}>
        <TopRow className={'top-row'}>
          {this.errorLine()}
          <TitleWrapper className={'title'}>
            <Ellipsify text={titleText} lines={2} />
          </TitleWrapper>
          {this.tickBox()}
        </TopRow>
        <BottomRow className={'bottom-row'}>
          <LeftColumn>{this.bottomLeftColumn()}</LeftColumn>
          <RightColumn>
            {actions ? (
              <CardActions
                actions={actions}
                onToggle={this.onMenuToggle}
                triggerColor={menuTriggerColor}
              />
            ) : null}
          </RightColumn>
        </BottomRow>
      </Overlay>
    );
  }

  errorLine() {
    const error = this.props.error;
    return (
      error && (
        <ErrorLine>
          <ErrorMessage>{this.props.error}</ErrorMessage>
        </ErrorLine>
      )
    );
  }

  tickBox() {
    const { selected, selectable } = this.props;
    const tick = <TickIcon label="tick" />;
    const className = cx('tickbox', { selected });

    return selectable && <TickBox className={className}> {tick} </TickBox>;
  }

  bottomLeftColumn() {
    const { error, onRetry } = this.props;

    if (error) {
      if (!onRetry) {
        return <ErrorIcon />;
      }

      return (
        <ErrorWrapper>
          <ErrorIcon />
          <Retry onClick={onRetry}>
            <FormattedMessage {...messages.retry} />
          </Retry>
        </ErrorWrapper>
      );
    } else {
      const { mediaType, subtitle, icon } = this.props;
      const classNames = cx('metadata');

      const fileIcon =
        mediaType || icon ? (
          <FileIcon mediaType={mediaType} iconUrl={icon} />
        ) : null;

      const subtitleEl = subtitle ? (
        <Subtitle className="file-size">{subtitle}</Subtitle>
      ) : null;

      return (
        <div>
          <Metadata className={classNames}>
            {fileIcon}
            {subtitleEl}
          </Metadata>
        </div>
      );
    }
  }

  onMenuToggle = (attrs: { isOpen: boolean }) => {
    this.setState({ isMenuExpanded: attrs.isOpen });
  };

  removeBtnClick(handler: CardEventHandler) {
    return (e: MouseEvent<HTMLDivElement>) => {
      e.preventDefault();
      e.stopPropagation();
      handler();
    };
  }
}
