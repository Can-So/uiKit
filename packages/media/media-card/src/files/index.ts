export * from './card';
export * from './cardImageView';
