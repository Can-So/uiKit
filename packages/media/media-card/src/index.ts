import { MouseEvent } from 'react';
import {
  MediaItemDetails,
  MediaCollectionItem,
  MediaType,
  FileProcessingStatus,
} from '@atlaskit/media-core';
import { UIAnalyticsEventInterface } from '@atlaskit/analytics-next-types';

import { CardAction } from './actions';

// the only components we expose to consumers is Card, CardView and CardList
export * from './root/card';
export * from './root/cardView';
export * from './root/domain';
export * from './list';
export * from './actions';
export { isUrlPreviewIdentifier } from './utils/identifier';
// TODO: don't expose this directly https://jira.atlassian.com/browse/FIL-4396
export {
  AppCardView,
  AppCardModel,
  convertAppCardToSmartCard,
} from './app_2/AppCardViewV2';

export type CardStatus =
  | 'uploading'
  | 'loading'
  | 'processing'
  | 'complete'
  | 'error';

export type CardAppearance =
  | 'auto'
  | 'small'
  | 'image'
  | 'square'
  | 'horizontal';

export type CardDimensionValue = number | string;

export interface CardDimensions {
  width?: CardDimensionValue;
  height?: CardDimensionValue;
}

export interface CardEvent {
  event: MouseEvent<HTMLElement>;
  mediaItemDetails?: MediaItemDetails;
}

export interface CardListEvent {
  event: MouseEvent<HTMLElement>;
  collectionName: string;
  mediaCollectionItem: MediaCollectionItem;
}

export interface OnSelectChangeFuncResult {
  selected: boolean;
  mediaItemDetails?: MediaItemDetails;
}

export interface OnSelectChangeFunc {
  (result: OnSelectChangeFuncResult): void;
}

export interface OnLoadingChangeState {
  readonly type: CardStatus;
  readonly payload?: Error | MediaItemDetails;
}

export interface OnLoadingChangeFunc {
  (state: OnLoadingChangeState): void;
}

export interface SharedCardProps {
  readonly appearance?: CardAppearance;
  readonly dimensions?: CardDimensions;

  readonly actions?: Array<CardAction>;
  readonly selectable?: boolean;
  readonly selected?: boolean;
}

export interface CardOnClickCallback {
  (result: CardEvent, analyticsEvent?: UIAnalyticsEventInterface): void;
}

export interface CardEventProps {
  readonly onClick?: CardOnClickCallback;
  readonly onMouseEnter?: (result: CardEvent) => void;
  readonly onSelectChange?: OnSelectChangeFunc;
  readonly onLoadingChange?: OnLoadingChangeFunc;
}

export interface AnalyticsFileAttributes {
  fileMediatype?: MediaType;
  fileMimetype?: string;
  fileStatus?: FileProcessingStatus;
  fileSize?: number;
}

export interface AnalyticsLinkAttributes {
  linkDomain: string;
}

export interface AnalyticsViewAttributes {
  viewPreview: boolean;
  viewActionmenu: boolean;
  viewSize?: CardAppearance;
}

export interface BaseAnalyticsContext {
  // These fields are requested to be in all UI events. See guidelines:
  // https://extranet.atlassian.com/display/PData/UI+Events
  packageVersion: string; // string â€” in a format like '3.2.1'
  packageName: string;
  componentName: string;

  actionSubject: string; // ex. MediaCard
  actionSubjectId: string | null; // file/link id
}

export interface CardAnalyticsContext extends BaseAnalyticsContext {}

export interface CardViewAnalyticsContext extends BaseAnalyticsContext {
  loadStatus: 'fail' | 'loading_metadata' | 'uploading' | 'complete';
  type: 'file' | 'link' | 'smart';
  viewAttributes: AnalyticsViewAttributes;
  fileAttributes?: AnalyticsFileAttributes;
  linkAttributes?: AnalyticsLinkAttributes;
}
