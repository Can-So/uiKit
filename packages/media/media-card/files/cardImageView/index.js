import * as tslib_1 from "tslib";
import * as React from 'react';
import { Component } from 'react';
import { CardOverlay } from './cardOverlay';
import { PlayIconWrapper, Wrapper, ProgressBarWrapper, Body, CardActionsWrapper, Overlay, ProgressWrapper, Title, } from './styled';
import { isLoadingImage } from '../../utils/isLoadingImage';
import { MediaImage } from '../../utils/mediaImage';
import { CardLoading } from '../../utils/cardLoading';
import VidPlayIcon from '@atlaskit/icon/glyph/vid-play';
import { shouldDisplayImageThumbnail } from '../../utils/shouldDisplayImageThumbnail';
import { Ellipsify } from '@atlaskit/media-ui';
import { ProgressBar } from '../../utils/progressBar';
import CardActions from '../../utils/cardActions';
var FileCardImageView = /** @class */ (function (_super) {
    tslib_1.__extends(FileCardImageView, _super);
    function FileCardImageView() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.renderCardContents = function () {
            var status = _this.props.status;
            if (status === 'error') {
                return _this.renderErrorContents();
            }
            else if (status === 'failed-processing') {
                return _this.renderFailedContents();
            }
            if (_this.isImageNotReadyForDisplay) {
                return _this.renderLoadingContents();
            }
            return _this.renderSuccessCardContents();
        };
        _this.renderLoadingContents = function () {
            return (React.createElement("div", { className: "wrapper" },
                React.createElement("div", { className: "img-wrapper" },
                    React.createElement(CardLoading, null))));
        };
        _this.renderErrorContents = function () {
            var _a = _this.props, error = _a.error, mediaName = _a.mediaName, mediaType = _a.mediaType, onRetry = _a.onRetry, actions = _a.actions, fileSize = _a.fileSize;
            return (React.createElement(React.Fragment, null,
                React.createElement("div", { className: "wrapper" }),
                React.createElement(CardOverlay, { persistent: true, mediaName: mediaName, mediaType: mediaType, error: error, onRetry: onRetry, actions: actions, subtitle: fileSize })));
        };
        _this.renderFailedContents = function () {
            var _a = _this.props, mediaName = _a.mediaName, mediaType = _a.mediaType, actions = _a.actions, fileSize = _a.fileSize;
            return (React.createElement(React.Fragment, null,
                React.createElement("div", { className: "wrapper" }),
                React.createElement(CardOverlay, { noHover: true, persistent: true, mediaName: mediaName, mediaType: mediaType, actions: actions, subtitle: fileSize })));
        };
        _this.renderUploadingCardOverlay = function () {
            var _a = _this.props, mediaType = _a.mediaType, dataURI = _a.dataURI, selectable = _a.selectable, selected = _a.selected;
            var isPersistent = mediaType === 'doc' || !dataURI;
            return (React.createElement(CardOverlay, { persistent: isPersistent, selectable: selectable, selected: selected }));
        };
        _this.renderPlayButton = function () {
            var mediaType = _this.props.mediaType;
            if (mediaType !== 'video') {
                return null;
            }
            return (React.createElement(PlayIconWrapper, null,
                React.createElement(VidPlayIcon, { label: "play", size: "large" })));
        };
        _this.renderMediaImage = function () {
            var _a = _this.props, dataURI = _a.dataURI, mediaType = _a.mediaType, previewOrientation = _a.previewOrientation;
            if (shouldDisplayImageThumbnail(dataURI, mediaType)) {
                return (React.createElement(MediaImage, { dataURI: dataURI, crop: _this.isCropped, stretch: _this.isStretched, previewOrientation: previewOrientation }));
            }
            return null;
        };
        _this.renderProgressBar = function () {
            var _a = _this.props, mediaName = _a.mediaName, progress = _a.progress, actions = _a.actions, status = _a.status;
            if (status !== 'uploading') {
                return null;
            }
            return (React.createElement(ProgressBarWrapper, null,
                React.createElement(Overlay, null,
                    React.createElement(Title, null,
                        React.createElement(Ellipsify, { text: mediaName || '', lines: 2 })),
                    React.createElement(Body, null,
                        React.createElement(ProgressWrapper, null,
                            React.createElement(ProgressBar, { progress: progress })),
                        React.createElement(CardActionsWrapper, null, actions ? (React.createElement(CardActions, { actions: actions, triggerColor: "white" })) : null)))));
        };
        _this.renderSuccessCardContents = function () {
            var _a = _this.props, disableOverlay = _a.disableOverlay, selectable = _a.selectable, status = _a.status;
            var overlay = null;
            if (!disableOverlay) {
                if (status === 'uploading') {
                    if (selectable) {
                        overlay = _this.renderUploadingCardOverlay();
                    }
                }
                else {
                    overlay = _this.renderSuccessCardOverlay();
                }
            }
            return (React.createElement("div", { className: "wrapper" },
                React.createElement("div", { className: "img-wrapper" },
                    _this.renderMediaImage(),
                    _this.renderProgressBar(),
                    _this.renderPlayButton()),
                overlay));
        };
        _this.renderSuccessCardOverlay = function () {
            var _a = _this.props, mediaName = _a.mediaName, mediaType = _a.mediaType, fileSize = _a.fileSize, dataURI = _a.dataURI, selectable = _a.selectable, selected = _a.selected, actions = _a.actions;
            var isPersistent = mediaType === 'doc' || !dataURI;
            return (React.createElement(CardOverlay, { persistent: isPersistent, selectable: selectable, selected: selected, mediaName: mediaName, mediaType: mediaType, subtitle: fileSize, actions: actions }));
        };
        return _this;
    }
    FileCardImageView.prototype.render = function () {
        var _a = this.props, disableOverlay = _a.disableOverlay, selectable = _a.selectable, selected = _a.selected, mediaType = _a.mediaType;
        return (React.createElement(Wrapper, { disableOverlay: disableOverlay, selectable: selectable, selected: selected, mediaType: mediaType }, this.renderCardContents()));
    };
    Object.defineProperty(FileCardImageView.prototype, "isImageNotReadyForDisplay", {
        get: function () {
            var _a = this.props, status = _a.status, dataURI = _a.dataURI, mediaType = _a.mediaType;
            if (dataURI) {
                return false;
            }
            return (status === 'loading' ||
                status === 'processing' ||
                isLoadingImage(mediaType, dataURI));
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FileCardImageView.prototype, "isCropped", {
        get: function () {
            var resizeMode = this.props.resizeMode;
            return resizeMode === 'crop';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FileCardImageView.prototype, "isStretched", {
        get: function () {
            var resizeMode = this.props.resizeMode;
            return resizeMode === 'stretchy-fit';
        },
        enumerable: true,
        configurable: true
    });
    FileCardImageView.defaultProps = {
        resizeMode: 'crop',
        disableOverlay: false,
    };
    return FileCardImageView;
}(Component));
export { FileCardImageView };
export default FileCardImageView;
//# sourceMappingURL=index.js.map