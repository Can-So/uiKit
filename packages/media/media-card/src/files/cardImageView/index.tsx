import * as React from 'react';
import { Component, ReactNode } from 'react';
import { MediaType, ImageResizeMode } from '@atlaskit/media-core';

import { CardDimensions, CardStatus } from '../../index';
import { CardAction } from '../../actions';

import { UploadingView } from '../../utils/uploadingView';
import { CardContent } from './cardContent';
import { CardOverlay } from './cardOverlay';
import { Wrapper } from './styled';
import { isLoadingImage } from '../../utils/isLoadingImage';

export interface FileCardImageViewProps {
  readonly mediaName?: string;
  readonly mediaType?: MediaType;
  readonly fileSize?: string;

  readonly dataURI?: string;
  readonly progress?: number;
  readonly status: CardStatus;

  readonly dimensions?: CardDimensions;
  readonly resizeMode?: ImageResizeMode;

  readonly disableOverlay?: boolean;
  readonly selectable?: boolean;
  readonly selected?: boolean;

  readonly error?: ReactNode;

  readonly actions?: Array<CardAction>;
  readonly onRetry?: () => void;
}

export class FileCardImageView extends Component<FileCardImageViewProps, {}> {
  static defaultProps = {
    resizeMode: 'crop',
    disableOverlay: false,
  };

  private isDownloadingOrProcessing() {
    const { status, dataURI, mediaType } = this.props;

    return (
      status === 'loading' ||
      status === 'processing' ||
      isLoadingImage(mediaType, dataURI)
    );
  }

  render() {
    const { disableOverlay, selectable, selected, mediaType } = this.props;
    return (
      <Wrapper
        disableOverlay={disableOverlay}
        selectable={selectable}
        selected={selected}
        mediaType={mediaType}
      >
        {this.getCardContents()}
      </Wrapper>
    );
  }

  private getCardContents = (): Array<JSX.Element> | JSX.Element => {
    const { status } = this.props;

    switch (status) {
      case 'error':
        return this.getErrorContents();
      case 'failed-processing':
        return this.getFailedContents();
      case 'uploading':
        return this.getUploadingContents();
      default:
        return this.getSuccessCardContents();
    }
  };

  private getErrorContents = (): JSX.Element => {
    const {
      error,
      mediaName,
      mediaType,
      onRetry,
      actions,
      fileSize,
    } = this.props;

    return (
      <>
        <div className="wrapper" />
        <CardOverlay
          persistent={true}
          mediaName={mediaName}
          mediaType={mediaType}
          error={error}
          onRetry={onRetry}
          actions={actions}
          subtitle={fileSize}
        />
      </>
    );
  };

  private getFailedContents = () => {
    const { mediaName, mediaType, actions, fileSize } = this.props;

    return (
      <>
        <div className="wrapper" />
        <CardOverlay
          noHover={true}
          persistent={true}
          mediaName={mediaName}
          mediaType={mediaType}
          actions={actions}
          subtitle={fileSize}
        />
      </>
    );
  };

  private getUploadingContents = (): JSX.Element => {
    const { actions, mediaName, progress, dataURI, selectable } = this.props;

    const overlay = selectable ? this.createUploadingCardOverlay() : null;

    return (
      <div className="wrapper">
        <div className="img-wrapper">
          <UploadingView
            title={mediaName}
            progress={progress || 0}
            dataURI={dataURI}
            actions={actions}
          />
        </div>
        {overlay}
      </div>
    );
  };

  private createUploadingCardOverlay = (): JSX.Element => {
    const { mediaType, dataURI, selectable, selected } = this.props;
    const isPersistent = mediaType === 'doc' || !dataURI;

    return (
      <CardOverlay
        persistent={isPersistent}
        selectable={selectable}
        selected={selected}
      />
    );
  };

  private getSuccessCardContents = (): JSX.Element => {
    const { mediaType, dataURI, disableOverlay } = this.props;
    const overlay =
      this.isDownloadingOrProcessing() || disableOverlay
        ? null
        : this.createSuccessCardOverlay();

    return (
      <div className="wrapper">
        <div className="img-wrapper">
          <CardContent
            loading={this.isDownloadingOrProcessing()}
            mediaItemType="file"
            mediaType={mediaType}
            dataURI={dataURI}
            crop={this.isCropped}
          />
        </div>
        {overlay}
      </div>
    );
  };

  private createSuccessCardOverlay = (): JSX.Element => {
    const {
      mediaName,
      mediaType,
      fileSize,
      dataURI,
      selectable,
      selected,
      actions,
    } = this.props;
    const isPersistent = mediaType === 'doc' || !dataURI;

    return (
      <CardOverlay
        persistent={isPersistent}
        selectable={selectable}
        selected={selected}
        mediaName={mediaName}
        mediaType={mediaType}
        subtitle={fileSize}
        actions={actions}
      />
    );
  };

  get isCropped() {
    const { resizeMode } = this.props;

    return resizeMode === 'crop';
  }
}

export default FileCardImageView;
