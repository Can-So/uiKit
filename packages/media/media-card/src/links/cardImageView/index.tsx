import * as React from 'react';
import { Component } from 'react';
import { ImageResizeMode } from '@atlaskit/media-core';

import { CardDimensions, CardAppearance, CardStatus } from '../../index';
import { CardImageView } from '../../utils/cardImageView';
import { CardAction } from '../../actions';

export interface LinkCardImageViewProps {
  linkUrl: string;
  title: string;
  site?: string;
  description?: string;
  thumbnailUrl?: string;
  iconUrl?: string;
  appearance?: CardAppearance;
  dimensions?: CardDimensions;
  resizeMode?: ImageResizeMode;
  status: CardStatus;
  actions?: Array<CardAction>;
  error?: string;
}

export class LinkCardImageView extends Component<LinkCardImageViewProps, {}> {
  render() {
    const {
      title,
      site,
      thumbnailUrl,
      status,
      dimensions,
      actions,
      error,
      iconUrl,
      linkUrl,
      resizeMode,
    } = this.props;

    return (
      <CardImageView
        mediaItemType="link"
        mediaName={title}
        subtitle={site || linkUrl}
        mediaType="image"
        dataURI={thumbnailUrl}
        status={status}
        dimensions={dimensions}
        actions={actions}
        error={error}
        icon={iconUrl}
        resizeMode={resizeMode}
      />
    );
  }
}
