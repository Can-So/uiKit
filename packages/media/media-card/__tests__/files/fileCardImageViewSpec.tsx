import * as React from 'react';
import { mount } from 'enzyme';

import { FileCardImageView } from '../../src/files';
import { CardOverlay } from '../../src/utils/cardImageView/cardOverlay';

describe('FileCardView', () => {
  it('should render card with non-persisting overlay when supplied mediaType is "image" and dataUri string is supplied', function() {
    const card = mount(
      <FileCardImageView mediaType="image" dataURI="data" status="complete" />,
    );
    expect(card.find(CardOverlay).props().persistent).toEqual(false);
  });

  it('should render empty wrapper when error prop is true', function() {
    const card = mount(
      <FileCardImageView error="Some random error occurred" status="error" />,
    );
    expect(card.find('.wrapper').children()).toHaveLength(0);
  });

  it('should render card overlay with the error prop true when supplied error prop is true', function() {
    const errorStr = 'Some random error occurred';
    const card = mount(<FileCardImageView error={errorStr} status="error" />);
    expect(card.find(CardOverlay).props().error).toEqual(errorStr);
  });

  it('should NOT render an overlay when loading prop is true', function() {
    const card = mount(<FileCardImageView status="loading" />);
    expect(card.find(CardOverlay)).toHaveLength(0);
  });
});
