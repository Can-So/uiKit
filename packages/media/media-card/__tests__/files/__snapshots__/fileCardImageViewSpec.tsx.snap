// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileCardView should render correct background based on mediaType" 1`] = `
.c0 {
  box-shadow: 0 1px 1px rgba(9,30,66,0.2),0 0 1px 0 rgba(9,30,66,0.24);
  cursor: pointer;
  background: #fff;
  line-height: normal;
  position: relative;
  width: 100%;
  height: 100%;
}

.c0 border-radius:function borderRadius() {
  return: 3;
}

.c0::after {
  content: '';
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  box-sizing: border-box;
  pointer-events: none;
  border: 2px solid transparent;
}

.c0::after border-radius:function borderRadius() {
  return: 3;
}

.c0 .wrapper {
  background: transparent;
  display: block;
  height: inherit;
  position: relative;
}

.c0 .wrapper border-radius:function borderRadius() {
  return: 3;
}

.c0 .wrapper .img-wrapper {
  position: relative;
  width: inherit;
  height: inherit;
  display: block;
  overflow: hidden;
}

.c0 .wrapper .img-wrapper border-radius:function borderRadius() {
  return: 3;
}

.c0 .wrapper .img-wrapper img {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  max-height: 100%;
  max-width: 100%;
  display: block;
}

<styled.div
  className="c0"
  mediaType="image"
/>
`;

exports[`FileCardView should render correct background based on mediaType" 2`] = `
.c0 {
  box-shadow: 0 1px 1px rgba(9,30,66,0.2),0 0 1px 0 rgba(9,30,66,0.24);
  cursor: pointer;
  background: #fff;
  line-height: normal;
  position: relative;
  width: 100%;
  height: 100%;
}

.c0 border-radius:function borderRadius() {
  return: 3;
}

.c0::after {
  content: '';
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  box-sizing: border-box;
  pointer-events: none;
  border: 2px solid transparent;
}

.c0::after border-radius:function borderRadius() {
  return: 3;
}

.c0 .wrapper {
  background: #F4F5F7;
  display: block;
  height: inherit;
  position: relative;
}

.c0 .wrapper border-radius:function borderRadius() {
  return: 3;
}

.c0 .wrapper .img-wrapper {
  position: relative;
  width: inherit;
  height: inherit;
  display: block;
  overflow: hidden;
}

.c0 .wrapper .img-wrapper border-radius:function borderRadius() {
  return: 3;
}

.c0 .wrapper .img-wrapper img {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  max-height: 100%;
  max-width: 100%;
  display: block;
}

<styled.div
  className="c0"
/>
`;
